receiver.py

```diff
@@ -72,13 +82,25 @@ def data_lines(self):
                         'source_addr_long', self.source_addr_long)
                 self.source_addr = packet.get(
                         'source_addr', self.source_addr)
+                self.rssi = packet.get('rssi', self.source_addr)
 
                 payload += packet['rf_data']
-                if x < self.expected_packets - 1 and len(payload) < 100:
-                    break;
 
-            # let our data be processed
-            yield self.data_shape.unpack(payload)
+                # Read first two bytes, to determine packet type
+                self.packet_type = int(payload[:2], 2)
```

Comment:
Also, not sure about using `int` to do this. I think this will fail on machines with other endian-ness.